src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -379,34 +373,35 @@ public async Task SaveChangesAsync(IEnumerable<UnspentOutputs> unspentOutputs, I
                         cacheItem.ExistInInner = unspentOutput != null;
                         cacheItem.IsDirty = false;
 
-                        cacheItem.UnspentOutputs = unspentOutput?.Clone();
-                        cacheItem.OriginalOutputs = unspentOutput?.Outputs.ToArray();
+                        cacheItem.UnspentOutputs = unspentOutput;
```

Comment:
Here I believe we are making the same again because we have having the same reference as the actual coin in the set that is used by coin view Rule